From 3513844ff4d3057e384c5b59604462938375285a Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Tue, 27 Sep 2022 12:50:08 +0200
Subject: [PATCH] arm64: dts: qcom: ipq6018: add QFPROM node

Add QFPROM node to IPQ6018 and expose the CPU speed bin only for now.

Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 arch/arm64/boot/dts/qcom/ipq6018.dtsi | 12 ++++++++++++
 1 file changed, 12 insertions(+)

--- a/arch/arm64/boot/dts/qcom/ipq6018.dtsi
+++ b/arch/arm64/boot/dts/qcom/ipq6018.dtsi
@@ -179,6 +179,18 @@
 		dma-ranges;
 		compatible = "simple-bus";
 
+		qfprom: efuse@a4000 {
+			compatible = "qcom,ipq6018-qfprom", "qcom,qfprom";
+			reg = <0x0 0xa4000 0x0 0x2000>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			cpu_speed_bin: cpu_speed_bin@135 {
+				reg = <0x135 0x1>;
+				bits = <7 1>;
+			};
+		};
+
 		prng: qrng@e1000 {
 			compatible = "qcom,prng-ee";
 			reg = <0x0 0xe3000 0x0 0x1000>;
